components/core/src/clp_s/ColumnWriter.hpp

```diff
@@ -34,9 +34,17 @@ class BaseColumnWriter {
     /**
      * Stores the column to a compressed file.
      * @param compressor
-     * @return the in-memory uncompressed size of the data written to the compressor
      */
-    virtual size_t store(ZstdCompressor& compressor) = 0;
+    virtual void store(ZstdCompressor& compressor) = 0;
+
+    /**
+     * Returns the total size of the header data that will be written to the compressor. This header
+     * size plus the sum of sizes returned by add_value is equal to the total size of data that will
```

Comment:
After the code change, `add_value` will not return any size?

Reply:
It does, but by reference. I'll change it to return by value so that it's less confusing.